test/integration/deployment/util.go

```diff
@@ -432,3 +432,66 @@ func (d *deploymentTester) scaleDeployment(newReplicas int32) error {
 	}
 	return nil
 }
+
+// waitForReadyReplicas waits for number of ready replicas to equal number of replicas.
+func (d *deploymentTester) waitForReadyReplicas() error {
+	if err := wait.PollImmediate(pollInterval, pollTimeout, func() (bool, error) {
+		deployment, err := d.c.ExtensionsV1beta1().Deployments(d.deployment.Namespace).Get(d.deployment.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, fmt.Errorf("failed to get deployment %q: %v", d.deployment.Name, err)
+		}
+		return deployment.Status.ReadyReplicas == *deployment.Spec.Replicas, nil
+	}); err != nil {
+		return fmt.Errorf("failed to wait for .readyReplicas to equal .replicas: %v", err)
+	}
+	return nil
+}
+
+// markUpdatedPodsReadyWithoutComplete marks updated Deployment pods as ready without waiting for deployment to complete.
+func (d *deploymentTester) markUpdatedPodsReadyWithoutComplete() error {
+	if err := wait.PollImmediate(pollInterval, pollTimeout, func() (bool, error) {
+		pods, err := d.listUpdatedPods()
+		if err != nil {
+			return false, err
+		}
+		for i := range pods {
+			pod := pods[i]
+			if podutil.IsPodReady(&pod) {
+				continue
+			}
+			if err = markPodReady(d.c, d.deployment.Namespace, &pod); err != nil {
+				d.t.Logf("failed to update Deployment pod %q, will retry later: %v", pod.Name, err)
+				return false, nil
+			}
+		}
+		return true, nil
+	}); err != nil {
+		return fmt.Errorf("failed to mark all updated pods as ready: %v", err)
+	}
+	return nil
+}
+
+// Verify all replicas fields of DeploymentStatus have desired count.
+// Immediately return an error when found a non-matching replicas field.
```

Comment:
When there is a mismatch, should we just print out an array of all 5 fields' value rather than returning an error mentioning _only_ the mismatched field?